### PR TITLE
Adding App Index helper

### PR DESCRIPTION
**Pull request by sojanpr**
Adding App Index helper to list app on google search

@aaustin

### CODE REVIEW

File: Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                    listOnGoogleSearch(urlForAppIndexing, context_, branchUniversalObject_);
+                } catch (Exception ignore) {
+                }
+            }
+        }
+    }
+
+    private static void listOnGoogleSearch(String shortLink, Context context, BranchUniversalObject branchUniversalObject) throws Exception {
+        // Create a Thing instance for the BUO with the link to the BUO
+        Class<?> ThingClass = Class.forName("com.google.android.gms.appindexing.Thing");
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java]: Didn't they move these out of Google's library and put them in Firebase's?

File: Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        Method setUrlMethod = ThingBuilderClass.getMethod("setUrl", Uri.class);
+        Method thingBuildMethod = ThingBuilderClass.getMethod("build");
+
+        setNameMethod.invoke(thingBuilder, branchUniversalObject.getTitle());
+        setDescMethod.invoke(thingBuilder, branchUniversalObject.getDescription());
+        setUrlMethod.invoke(thingBuilder, Uri.parse(shortLink));
+        Object thingObj = thingBuildMethod.invoke(thingBuilder);
+
+        // Now Create an View Action for the Thing created
+        Class<?> ThingActionClass = Class.forName("com.google.android.gms.appindexing.Action");
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java]: Same here

File: Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java [hunk 3]
```diff
@@ -242,6 +242,18 @@ public BranchUniversalObject setContentExpiration(Date expirationDate) {
...
 
     /**
+     * <p/>
+     * Publish this BUO with Google app indexing so that the contents will be available with gooole search
+     * with branch link pointing to the app.
+     * </p?>
+     *
+     * @param context Application context
+     */
+    public void listOnGoogleSearch(Context context) {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java]: Can you update the README as well with this?

File: Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java [hunk 4]
```diff
@@ -242,6 +242,18 @@ public BranchUniversalObject setContentExpiration(Date expirationDate) {
     }
 
     /**
+     * <p/>
+     * Publish this BUO with Google app indexing so that the contents will be available with gooole search
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java]: Typo

File: Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ * <p>
+ * Helper class for publishing BUO with google app indexing
+ * </p>
+ */
+class AppIndexingHelper {
+    public static void addToAppIndex(final Context context, final BranchUniversalObject buo) {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                String urlForAppIndexing = buo.getShortUrl(context, new LinkProperties());
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/indexing/AppIndexingHelper.java]: let's prefill the link properties similar to iOS: `[linkData setupChannel:@"google_search"];`

**Comment by aaustin** [comment]
👍

File: Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java (hunk 3 above)
Comment (aeromusek) [inline Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java]: 'gooole' is misspelled in the comment

Comment (aeromusek) [inline Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java]: My bad, sorry. Missed the later commit

**Pull request merged**